### ISSUE TITLE
Changing column width in JabRef 5.0 prevents opening 4.3.1

### ISSUE THREAD

**Issue by douglasrizzo** [JabRef/jabref#4579]
JabRef version 5.0 and 4.3.1

- [X] I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists

1. Open JabRef 5.0 and add or change at least one of the column widths in _Preferences -> Entry table columns_.
2. Save preferences.
3. Open JabRef 4.3.1.
4. Open JabRef 4.3.1 Preferences window.

**Expected behavior:** the preferences window in JabRef 4.3.1 opens.

**Actual behavior:** the preferences window in JabRef 4.3.1 does not open. An error log in my terminal points out that every column width for the main table is saved in an unknown or wrong format (e.g "100.0", "32.0" etc).

**Workaround:** open JabRef 5.0, delete all entries in Entry table columns, open JabRef 4.3.1 again and the Preferences should work.

**Why it matters:** I've been testing version 5.0, but I still face some problems that force me to use version 4.3.1 when doing actual work. This bug makes it impossible to use both versions, unless there is a way to separate their preferences that I don't know about.

**tl;dr:** JabRef 5 treats the column width values of the main table as floats, JabRef < 5 treats them as integers. This makes opening the preferences in older JabRef after editing them in new JabRef impossible.

**Comment by Siedlerchr** [comment]
You could export your preferences or use the memory stick mode

**Comment by douglasrizzo** [comment]
I just found out about both of these modes. I don't know what to make of this incompatibility, though. Maybe the major version change from 4 to 5 justifies it...

**Comment by Siedlerchr** [comment]
Due to lack of resources (aka developers) we can usually not provide backwards compatibility. Especially with the transition of the UI to JavaFX we can't avoid breaking changes somewhere. 
We welcome every Java developer. 
And of course we rely on feedback from users, if things work properly on all platforms. 

You can export your preferences, change the values to integer and import them back in.

**Comment by douglasrizzo** [comment]
In order to import preferences in JabRef 4.3.1, I'd need to be able to open the Preferences window, which I can't do if the current values are already floats. What I think would work is changing the values in `~/.java/.userPrefs/org/jabref/prefs.xml` (I'm on Linux).

**Comment by tobiasdiez** [comment]
While in general we don't have the resources to think too much about backwards migration, here, a bit of work on our side is probably the best solution. We just need to convert the double column widths returned by JavaFX to integers before saving them in the preferences. Not much work and improved user experience.

**Comment by wujastyk** [comment]
JabRef 5.0-dev--snapshot--2019-01-22--master--05f3350cc
Linux 4.15.0-43-generic amd64 
Java 1.8.0_201

I'm in the same situation: 4.3.1 for serious work, 5 for testing.  I can't open "preferences " in 4.3.1 any longer.

I see another problem that is probably related to the above issue.  In 5, I can go to preferences and change the column sequence and spacing (manually or update to current), but my changes are never saved.

**Comment by wujastyk** [comment]
Confirmed fix.  Thank you!

**Comment by BreakBB** [comment]
An ugly workaround for this is to [reset preferences](http://help.jabref.org/en/CommandLine#reset-preferences--d-key) using the command line.

**Comment by Siedlerchr** [comment]
This is now fixed in the latest master version. Meaning that the latest master from now on stores the column widths as integers again and therefore older JabRef versions can access the prefs again.,

We would like to ask you to use a development build from https://builds.jabref.org/master and report back if it works for you. Please remember to make a backup of your library before trying-out this version.
